fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientsApiResource.java

```diff
@@ -415,25 +300,302 @@ public String postClientTemplate(@QueryParam("legalFormType") final String legal
+            @QueryParam("command") @Parameter(description = "command") final String commandParam,
+            @Parameter(hidden = true) final String apiRequestBodyAsJson) {
+        return applyCommandOverClient(null, externalId, commandParam, apiRequestBodyAsJson);
+    }
+
+    @DELETE
+    @Path("/external-id/{externalId}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Delete a Client", description = "If a client is in Pending state, you are allowed to Delete it. The delete is a 'hard delete' and cannot be recovered from. Once clients become active or have loans or savings associated with them, you cannot delete the client but you may Close the client if they have left the program.")
+    @RequestBody(required = true, content = @Content(schema = @Schema(implementation = ClientsApiResourceSwagger.DeleteClientsClientIdRequest.class)))
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = ClientsApiResourceSwagger.DeleteClientsClientIdResponse.class))) })
+    public String delete(@PathParam("externalId") @Parameter(description = "externalId") final String externalId) {
+        return deleteClient(null, externalId);
+    }
+
+    @GET
+    @Path("/external-id/{externalId}/obligeedetails")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Retrieve client obligee details", description = "Retrieve client obligee details using the client external Id")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = ClientsApiResourceSwagger.GetClientObligeeDetailsResponse.class))),
+            @ApiResponse(responseCode = "400", description = "Bad Request") })
+    public String retrieveObligeeDetails(@PathParam("externalId") final String externalId, @Context final UriInfo uriInfo) {
+        return retrieveClientObligeeDetails(null, externalId);
+    }
 
+    @GET
+    @Path("/external-id/{externalId}/transferproposaldate")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Retrieve client transfer template", description = "Retrieve client transfer template using the client external Id")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = ClientsApiResourceSwagger.GetClientTransferProposalDateResponse.class))),
+            @ApiResponse(responseCode = "400", description = "Bad Request") })
+    public String retrieveTransferTemplate(@PathParam("externalId") final String externalId, @Context final UriInfo uriInfo) {
+        return retrieveClientTransferTemplate(null, externalId);
+    }
+
+    public String retrieveAll(final UriInfo uriInfo, final String sqlSearch, final Long officeId, final String externalId,
```

Comment:
This can be private

Reply:
No, It can not be private, because It is being used by the Self service API resource

<img width="1099" alt="Screenshot 2022-12-06 at 8 41 14" src="https://user-images.githubusercontent.com/44206706/205941913-81052e7e-4a00-42c7-b01f-a93412716695.png">